### PR TITLE
Confusion matrix component

### PR DESCRIPTION
**Pull request by porterjc**
I think I got most of the commented code cleaned up as well as unused imports, but I may have missed some. Same thing with Javadocs, let us know if we missed any.

### CODE REVIEW

File: src/main/java/com/metsci/laproc/ROCCurvePlot.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public SimplePlot2D getLayout()
         Axis yAxis = graph.getYAxis();
 
         // set axis labels and chart title
-        plot.setTitle( "ROC Curve Example" );
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/ROCCurvePlot.java]: This should probably say Curve not SimpleGraphableData (silly intellij refactoring)

File: src/main/java/com/metsci/laproc/data/ClassifierDataSet.java [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+
+    /**
+     * Adds a point to the data set
+     * @param point The point to add
+     */
+    public void add(DataPoint point) {
+        this.data.add(point);
+    }
+
+    public Set<DataPoint> getAllPoints() {
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/data/ClassifierDataSet.java]: JavaDocs looks like they got kinda moved around here

File: src/main/java/com/metsci/laproc/plotting/GraphableFunction.java [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
  * An interface representing any computable function
  * Created by robinsat on 9/20/2016.
  */
 public interface GraphableFunction {
 
-    /** Executes the function and returns the resulting data set */
-    public GraphableFunctionOutput compute();
+    /**
+     * Executes the function and returns the resulting data set
+     * @return
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/plotting/GraphableFunction.java]: This @return isnt filled out, probably just @return resulting dataset

File: src/main/java/com/metsci/laproc/plotting/ROCCurve.java [hunk 4]
```diff
@@ -37,21 +39,21 @@ public ROCCurve(ClassifierDataSet in) {
...
         out.addAttribute(fnrString);
         out.addAttribute(cutpointString);
-        out.addAttribute(accuracyString);
+        out.addAttribute(accuracyString); */
 
         // Calculate the number of positive values and negative values in this data set
         int numPositives = 0;
         int numNegatives = 0;
-        for(DataPoint p : input) {
+        for(com.metsci.laproc.data.DataPoint p : input) {
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/plotting/ROCCurve.java]: Whats up with this? Why didn't importing work?

File: src/main/java/com/metsci/laproc/plotting/ROCCurve.java [hunk 5]
```diff
@@ -37,21 +39,21 @@ public ROCCurve(ClassifierDataSet in) {
...
      * @return The plottable data set representing this curve
      */
-    public GraphableFunctionOutput compute() {
-        GraphableFunctionOutput out = new GraphableFunctionOutputImpl();
+    public GraphableData compute() {
+        GraphableDataWithStats out = new GraphableDataWithStats();
 
         // Add the attributes to the list
-        out.addAttribute(tprString);
+      /*  out.addAttribute(tprString);
```
Comment (christophermalinosky) [inline src/main/java/com/metsci/laproc/plotting/ROCCurve.java]: Can probably delete this

**Comment by porterjc** [comment]
Updated code to fix the javadocs and some of the other odd problems.

File: src/main/java/com/metsci/laproc/plotting/ROCCurve.java (hunk 4 above)
Comment (porterjc) [inline src/main/java/com/metsci/laproc/plotting/ROCCurve.java]: This was alt+enter deciding to do really convoluted things.

**Pull request merged**